docs/sources/operators-guide/configure/configuring-hash-rings.md

```diff
@@ -77,6 +77,25 @@ If you run multiple Mimir processes on the same node or the port `7946` is not a
 - `-memberlist.advertise-addr`: IP address to advertise to other Mimir replicas. The other replicas will connect to this IP to talk to the instance.
 - `-memberlist.advertise-port`: Port to advertise to other Mimir replicas. The other replicas will connect to this port to talk to the instance.
 
+#### Cluster label verification
+
+By default, Grafana Mimir memberlist will join a cluster with any instance discovered when resolving the hosts configured via the `-memberlist.join` CLI flag (or its respective YAML configuration option).
+If, for any reason, the discovered addresses include instances of other Grafana Mimir clusters, or instances of other distributed systems using memberlist, Grafana Mimir will join these unrelated clusters together.
+
+To avoid this type of undesired situation Grafana Mimir offers an additional type of validation known as cluster label verification.
+
+When enabled, all memberlist internal traffic will be prefixed with the configured cluster label, discarding any other traffic that doesn't match that prefix and ensuring that only replicas having the same configured label can connect to each other.
+
+#### Migrate to using cluster label verification
+
+To migrate a Grafana Mimir cluster to use cluster label verification, the following steps need to be performed:
+
+- Disable cluster label verification on all cluster instances via `-memberlist.cluster-label-verification-disabled` CLI flag (or its respective YAML configuration option).
+- Define a label on all cluster instances which is unique to the cluster via `-memberlist.cluster-label` CLI flag (or its respective YAML configuration option).
+  This label must be the same on all instances that are part of the same cluster.
+  For instance, some value like `<k8s cluster>.<k8s namespace>` would make sense.
```

Comment:
It's not safe to give this suggestion. For example, I think with the Helm chart you could have 2 Mimir clusters in the same namespace. What you could say is "For instance, if you run a Mimir cluster in a dedicated namespace, then setting the cluster label to the name of the namespace should be a safe choice."